samples/net/gsm_modem/src/main.c

```diff
@@ -6,15 +6,59 @@
 
 #include <zephyr.h>
 #include <sys/printk.h>
+#include <shell/shell.h>
 #include <drivers/uart.h>
 #include <net/net_mgmt.h>
 #include <net/net_event.h>
 #include <net/net_conn_mgr.h>
+#include <drivers/gsm_ppp.h>
 
 #include <logging/log.h>
 LOG_MODULE_REGISTER(sample_gsm_ppp, LOG_LEVEL_DBG);
 
+static const struct device *gsm_dev;
 static struct net_mgmt_event_callback mgmt_cb;
+static bool connected;
+
+static int cmd_sample_modem_suspend(const struct shell *shell,
+				    size_t argc, char *argv[])
+{
+	if (!connected) {
+		LOG_INF("Not connected.");
```

Comment:
Why not use `shell_print(shell, "Not connected.");` ?

Reply:
Yep, that would actually be a more proper print, I just copied this from other code thus logging print was used.
Anyway, this commit does not really need to be part of this PR, it just helps when testing/debugging the feature.
@endian-benjamin I think we need to first get #28475 merged as it and this PR contain shared commits, and then come back to this fix.

Reply:
I think you're missing the fact that this is based on an unmerged PR :)

Perhaps this should be marked WIP or something until it's merged? What's the procedure?